lib/elixir/pages/anti-patterns/macro-anti-patterns.md

```diff
@@ -4,11 +4,106 @@ This document outlines anti-patterns related to meta-programming.
 
 ## Unnecessary macros
 
-TODO.
+#### Problem
+
+`Macros` are powerful meta-programming mechanisms that can be used in Elixir to extend the language. While implementing `macros` is not an anti-pattern in itself, this meta-programming mechanism should only be used when absolutely necessary. Whenever a `macros` is implemented, and it was possible to solve the same problem using functions or other pre-existing Elixir structures, the code becomes unnecessarily more complex and less readable. Because `macros` are more difficult to implement and understand, their indiscriminate use can compromise the evolution of a system, reducing its maintainability.
+
+#### Example
+
+The `MyMath` module implements the `sum/2` macro to perform the sum of two numbers received as parameters. While this code has no syntax errors and can be executed correctly to get the desired result, it is unnecessarily more complex. By implementing this functionality as a macro rather than a conventional function, the code became less clear and less objective:
```

Comment:
```suggestion
The `MyMath` module implements the `sum/2` macro to perform the sum of two numbers received as parameters. While this code has no syntax errors and can be executed correctly to get the desired result, it is unnecessarily more complex. By implementing this functionality as a macro rather than a conventional function, the code became less clear:
```